openfeature-provider-local/src/main/java/com/spotify/confidence/OpenFeatureLocalResolveProvider.java

```diff
@@ -0,0 +1,198 @@
+package com.spotify.confidence;
+
+import com.google.protobuf.Struct;
+import com.spotify.confidence.flags.resolver.FlagResolverService;
+import com.spotify.confidence.flags.resolver.SidecarResolverServiceFactory;
+import com.spotify.confidence.shaded.flags.resolver.v1.ResolveFlagsRequest;
+import com.spotify.confidence.shaded.flags.resolver.v1.ResolveFlagsResponse;
+import com.spotify.confidence.shaded.flags.resolver.v1.ResolvedFlag;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.FeatureProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.FlagNotFoundError;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import org.slf4j.Logger;
+
+public class OpenFeatureLocalResolveProvider implements FeatureProvider {
+  private final String clientSecret;
+  private static final Logger log =
+      org.slf4j.LoggerFactory.getLogger(OpenFeatureLocalResolveProvider.class);
+  private final FlagResolverService flagResolverService;
+
+  public OpenFeatureLocalResolveProvider(ApiSecret apiSecret, String clientSecret) {
+    this.flagResolverService = SidecarResolverServiceFactory.from(apiSecret, clientSecret);
```

Comment:
What happens between instantiating this object and getting the first state? Do we supported ways to wait for the provider to be ready, perhaps according to the OpenFeature specifications?